lib/rack/profiler.rb

```diff
@@ -76,19 +82,19 @@ def render_profiler_results(env)
       ActiveSupport::Notifications.instrument('rack-profiler.total_time') do
         status, headers, body = @app.call(env)
       end
-      [ 200,
-        { 'Content-Type' => 'application/json' },
-        [ { events:   events.sort_by { |event| event[:start] },
-            response: {
-            status:  status,
-            headers: headers,
-            body:    stringify_body(body)
-          }
-        }.to_json ]
-      ]
+      return [status, headers, body] unless authorized?(env)
+      results = {
+        events:   events.sort_by { |event| event[:start] },
+        response: {
+          status:  status,
+          headers: headers,
+          body:    stringify_body(body)
+        }
+      }
+      [200, { 'Content-Type' => 'application/json' }, [results.to_json]]
     end
 
-    def render_dashboard
+    def render_dashboard(env)
```

Comment:
Unused method argument - `env`. If it's necessary, use `_` or `_env` as an argument name to indicate that it won't be used. You can also write as `render_dashboard(*)` if you want the method to accept any arguments but don't care about them.